docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,323 @@
+.. _Courses API Blocks Resource:
+
+########################################
+Courses API Blocks Resource
+########################################
+
+With the Courses API **Blocks** resource, you can complete the
+following tasks.
+
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _Get a list of the course blocks in a course:
+
+****************************************
+Get a List of Course Blocks in a Course
+****************************************
+
+The endpoint to get a list of course blocks in a course is
+``/api/courses/v1/blocks/`` with required ``course_id`` parameter.
+
+=====================
+Use Case
+=====================
+
+Get a list of course blocks in a specified course. Response results depend on
+the requesting user's access level.
+
+=====================
+Example Requests
+=====================
+
+GET /api/courses/v1/blocks/?course_id=<course_id>
+
+GET /api/courses/v1/blocks/?course_id=edX%2FDemoX%2FDemo_Course
+&all_blocks=true&requested_fields=graded,format,student_view_multi_device
+
+=====================
+Query Parameters
+=====================
+
+* all_blocks: (boolean) Provide a value of ``true`` to return all blocks.
+
+* block_counts: (list) Specify the types of blocks for which to return a count
+  of that type (including child blocks). Example: ``block_counts=video,problem``.
+
+* block_types_filter: (list) Specify the types of blocks to be included in results.
+  Possible values are: ``sequential``, ``vertical``, ``html``, ``problem``,
+  ``video`` and ``discussion``. Example: ``block_types_filter=problem,html``.
+
+* course_id: (string, required) The URL-encoded ID of the course whose block
+  data you are requesting. Example: ``course_id=edX%2FDemoX%2FDemo_Course``.
+
+* depth: (integer or ``all``) Specify how far in the course blocks hierarchy
+  to traverse down. A value of ``all`` specifies the entire hierarchy. The
+  default value is ``0``. Example: ``depth=all``.
+
+* requested_fields: (list) Specify the fields to return for each block, in
+  addition to ID, type, and display_name, which are always returned. For the
+  list of possible fields, see the fields listed under ``blocks`` in the
+  **Response Values** section below. Example:
+  ``requested_fields=graded,format,student_view_multi_device``.
+
+* return_type: (string) Specify the data type in which the block data is
+  returned. Supported values are ``dict`` and ``list``. The default value is
+  ``dict``.
+
+* student_view_data: (list) Specify for which block types the JSON data is
+  returned in the "student_view_data" response value. Example:
+  ``student_view_data=video``.
+
+* username: (string) Required, unless ``all_blocks`` is specified. Example:
```

Comment:
The username indicates for which user the blocks are returned, and the API returns user-specific content for that user.  This includes group-affiliated content and user-assigned content.  Staff users are allowed to specify other users' usernames.